test/functional/api/cas/statistics.py

```diff
@@ -0,0 +1,643 @@
+            and self.free == other.free
+            and self.clean == other.clean
+            and self.dirty == other.dirty
+        )
+
+
+class InactiveUsageStats:
+    def __init__(self, inactive_occupancy, inactive_clean, inactive_dirty):
+        self.inactive_occupancy = inactive_occupancy
+        self.inactive_clean = inactive_clean
+        self.inactive_dirty = inactive_dirty
+
+    def __str__(self):
+        return (
+            f"Inactive usage stats:\n"
+            f"Inactive occupancy: {self.inactive_occupancy}\n"
+            f"Inactive clean: {self.inactive_clean}\n"
+            f"Inactive dirty: {self.inactive_dirty}"
+        )
+
+    def __eq__(self, other):
+        if not other:
+            return False
+        return (
+            self.inactive_occupancy == other.inactive_occupancy
+            and self.inactive_clean == other.inactive_clean
+            and self.inactive_dirty == other.inactive_dirty
+        )
+
+
+class RequestStats:
+    def __init__(
+        self,
+        read_hits,
+        read_part_misses,
+        read_full_misses,
+        read_total,
+        write_hits,
+        write_part_misses,
+        write_full_misses,
+        write_total,
+        pass_through_reads,
+        pass_through_writes,
+        requests_serviced,
+        requests_total,
+    ):
+        self.read = RequestStatsChunk(
+            read_hits, read_part_misses, read_full_misses, read_total
+        )
+        self.write = RequestStatsChunk(
+            write_hits, write_part_misses, write_full_misses, write_total
+        )
+        self.pass_through_reads = pass_through_reads
+        self.pass_through_writes = pass_through_writes
+        self.requests_serviced = requests_serviced
+        self.requests_total = requests_total
+
+    def __str__(self):
+        return (
+            f"Request stats:\n"
+            f"Read: {self.read}\n"
+            f"Write: {self.write}\n"
+            f"Pass-through reads: {self.pass_through_reads}\n"
+            f"Pass-through writes: {self.pass_through_writes}\n"
+            f"Serviced requests: {self.requests_serviced}\n"
+            f"Total requests: {self.requests_total}"
+        )
+
+    def __eq__(self, other):
+        if not other:
+            return False
+        return (
+            self.read == other.read
+            and self.write == other.write
+            and self.pass_through_reads == other.pass_through_reads
+            and self.pass_through_writes == other.pass_through_writes
+            and self.requests_serviced == other.requests_serviced
+            and self.requests_total == other.requests_total
+        )
+
+
+class RequestStatsChunk:
+    def __init__(self, hits, part_misses, full_misses, total):
+        self.hits = hits
+        self.part_misses = part_misses
+        self.full_misses = full_misses
+        self.total = total
+
+    def __str__(self):
+        return (
+            f"\nHits: {self.hits}\n"
```

Comment:
I would remove `\n` at the beginning of string, and leave it to be printed by caller.